indico/modules/events/persons/controllers.py

```diff
@@ -382,6 +385,23 @@ def _process(self):
         return jsonify_form(form)
 
 
+class RHDeleteUnusedEventPerson(RHPersonsBase):
+    def _process_args(self):
+        RHPersonsBase._process_args(self)
+        self.person = EventPerson.query.with_parent(self.event).filter_by(id=request.view_args['person_id']).one()
+
+    def _is_unused(self):
+        persons = self.get_persons()
+        person = persons.get(self.person.email or self.person.id)
+        return not person or person['roles'].get('no_roles', False)
```

Comment:
I'm not a big fan of this logic here. It's fine to determine whether or not to show the delete option, but for the actual deletion here I'd rather add a `has_links()` method to EventPerson that checks the various relationships backrefs. I think they are all lazy=dynamic so you can use `.has_rows()` on them.

(it'll also help us spotting cases where we are missing some role on this page, because then the deletion would fail with an error and the user may report that)